include/libtorrent/torrent_info.hpp

```diff
@@ -131,7 +131,7 @@ namespace aux {
 	{
 		int max_buffer_size = 6000000;
 		// the max number of pieces allowed in the torrent
-		int max_pieces = 0x100000;
```

Comment:
Is there a reason the tool you use can't be changed instead, to use a higher limit?
That's the point of having `load_torrent_limits` in the first place, to let clients configure it. Whether that limit is exposed to the *user* is a different issue, if you just want to hard code it, it doesn't have to be.

Reply:
The limit is still present and still defends against DOS.  The change provides default support for more torrents, for users unfamiliar with adjusting it.  Do you have a concern with it?

Reply:
right, it's not obvious that it "helps" the majority of clients though. It might "break" most of the users (who knows). Any computer that can't afford to allocate 30 megabytes for a .torrent file would have an issue with this patch.

Reply:
what devices does libtorrent target that can't allocate 30MB?

Honestly the torrent I was worried about people not being able to access only needs a very small bump to the old value.  I tried to pick a value that seemed reasonable for use cases I was familiar with.  Would you accept a smaller increase?

Reply:
> what devices does libtorrent target that can't allocate 30MB?

I'm not convinced that that's the most relevant question. Any device will have degraded performance by allocating more memory than necessary. I would think a better question would be; "what is the 99th percentile of torrent sizes in the wild?", or maybe something along those lines.

basically, at what point is it likely the torrent is malicious. But also allocating a relatively large amount of memory is of course relevant too.

> Would you accept a smaller increase?

I would like a stronger argument why not each client can pick this limit themselves. It would seem the client is in 
a much better position to judge what reasonable limits are. Also, these limits only apply to certain "convenience" constructors of `torrent_info`, ones that loads from disk and parses bencoding. It's not limits that apply broadly to all aspects of libtorrent.